tests/pipelines/test_pipelines_common.py

```diff
@@ -1373,6 +1373,7 @@ def test_sequential_cpu_offload_forward_pass(self, expected_max_diff=1e-4):
         output_without_offload = pipe(**inputs)[0]
 
         pipe.enable_sequential_cpu_offload()
+        assert pipe._execution_device.type == pipe._offload_device.type
```

Comment:
add a check to make sure the `_execution_device` attribute of pipeline work as expected since we rely on the accelerate hooks to infer the device https://github.com/huggingface/diffusers/blob/56bd7e67c2e01122cc93d98f5bd114f9312a5cce/src/diffusers/pipelines/pipeline_utils.py#L981

Reply:
Does it also make sense to check if the `SequentialHook` is properly installed when a model in a pipeline has buffers?